plugins/bundle/plugin.go

```diff
@@ -301,25 +305,45 @@ func (p *Plugin) initDownloaders() {
 	}
 }
 
-func (p *Plugin) loadAndActivateBundlesFromDisk(ctx context.Context) error {
+func (p *Plugin) loadAndActivateBundlesFromDisk(ctx context.Context) {
+
+	persistedBundles := map[string]*bundle.Bundle{}
+
 	for name, src := range p.config.Bundles {
 		if p.persistBundle(name) {
 			b, err := loadBundleFromDisk(p.bundlePersistPath, name, src)
 			if err != nil {
 				p.log(name).Error("Failed to load bundle from disk: %v", err)
-				return err
+				p.status[name].SetError(err)
+				continue
 			}
 
 			if b == nil {
-				return nil
+				continue
 			}
 
+			persistedBundles[name] = b
+		}
+	}
+
+	if len(persistedBundles) == 0 {
+		return
+	}
+
+	for retry := 0; ; retry++ {
+		if retry >= maxActivationRetry {
```

Comment:
This looks a little odd, can we do this? (I might be missing something here)
```suggestion
	for retry := 0; retry < maxActivationRetry ; retry++ {
```

Reply:
Updated